include/openssl/kdf.h

```diff
@@ -15,80 +15,104 @@
 extern "C" {
 #endif
 
-# define EVP_PKEY_CTRL_TLS_MD                   (EVP_PKEY_ALG_CTRL)
+/* Generic controls */
+# define EVP_PKEY_CTRL_KDF_MD                   (EVP_PKEY_ALG_CTRL)
+# define EVP_PKEY_CTRL_KDF_PASS                 (EVP_PKEY_ALG_CTRL + 8)
+# define EVP_PKEY_CTRL_KDF_SALT                 (EVP_PKEY_ALG_CTRL + 4)
+# define EVP_PKEY_CTRL_KDF_KEY                  (EVP_PKEY_ALG_CTRL + 5)
+# define EVP_PKEY_CTRL_KDF_INFO                 (EVP_PKEY_ALG_CTRL + 6)
+# define EVP_PKEY_CTRL_KDF_MODE                 (EVP_PKEY_ALG_CTRL + 7)
+
+/* KDF-specific controls */
 # define EVP_PKEY_CTRL_TLS_SECRET               (EVP_PKEY_ALG_CTRL + 1)
 # define EVP_PKEY_CTRL_TLS_SEED                 (EVP_PKEY_ALG_CTRL + 2)
-# define EVP_PKEY_CTRL_HKDF_MD                  (EVP_PKEY_ALG_CTRL + 3)
-# define EVP_PKEY_CTRL_HKDF_SALT                (EVP_PKEY_ALG_CTRL + 4)
-# define EVP_PKEY_CTRL_HKDF_KEY                 (EVP_PKEY_ALG_CTRL + 5)
-# define EVP_PKEY_CTRL_HKDF_INFO                (EVP_PKEY_ALG_CTRL + 6)
-# define EVP_PKEY_CTRL_HKDF_MODE                (EVP_PKEY_ALG_CTRL + 7)
-# define EVP_PKEY_CTRL_PASS                     (EVP_PKEY_ALG_CTRL + 8)
-# define EVP_PKEY_CTRL_SCRYPT_SALT              (EVP_PKEY_ALG_CTRL + 9)
 # define EVP_PKEY_CTRL_SCRYPT_N                 (EVP_PKEY_ALG_CTRL + 10)
 # define EVP_PKEY_CTRL_SCRYPT_R                 (EVP_PKEY_ALG_CTRL + 11)
 # define EVP_PKEY_CTRL_SCRYPT_P                 (EVP_PKEY_ALG_CTRL + 12)
 # define EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES      (EVP_PKEY_ALG_CTRL + 13)
 
+/* Aliases for backwards-compatibility */
+# define EVP_PKEY_CTRL_TLS_MD                   EVP_PKEY_CTRL_KDF_MD
+# define EVP_PKEY_CTRL_HKDF_MD                  EVP_PKEY_CTRL_KDF_MD
```

Comment:
Hi @levitte, thanks for the comprehensive review; I've just pushed ecd0eb6d which should address all of your issues. I've opted for option #2 (it makes it very clear to the user that they're not supposed to use it anymore).